boards/arm/arduino_giga_r1/arduino_giga_r1_m4.dts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/h7/stm32h747Xi_m4.dtsi>
+#include <st/h7/stm32h747xihx-pinctrl.dtsi>
+#include "arduino_giga_r1.dtsi"
+
+/ {
+	model = "Arduino GIGA R1 WiFi Board (M4)";
+	compatible = "arduino,giga-r1";
+
+	chosen {
+		zephyr,console = &usart2;
+		zephyr,shell-uart = &usart2;
+		zephyr,sram = &sram1;
+		zephyr,flash = &flash1;
```

Comment:
Where does the code for this reside? i.e. `zephyr,code-partition =` ?

Reply:
`flash1`, the MCU has two independent flash banks, in the default config the M7 core entry point is the start of `flash0` and the M4 one is `flash1`, I think that way they get unblocked access through the bus matrix, see:

![x](https://user-images.githubusercontent.com/891546/222754183-2739e26a-ac3a-4be5-8329-d9a002459c02.png)

It's really the same settings as `stm32h747i_disco`, good enough for running some samples but realistically if ones wants to use the second core properly there's more custom setup to do anyway.